net.sourceforge.xournal.json

```diff
@@ -0,0 +1,80 @@
+{
+  "app-id": "net.sourceforge.xournal",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.28",
+  "sdk": "org.gnome.Sdk",
+  "command": "xournal",
+  "finish-args": [
+    "--share=ipc",
+    "--socket=x11",
+    "--filesystem=host"
+  ],
+  "modules": [
+    {
+      "name": "libart",
+      "cleanup": [
+        "/bin"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://download.gnome.org/sources/libart_lgpl/2.3/libart_lgpl-2.3.21.tar.gz",
+          "sha256": "2a10b1c884bb37c7bd5c5eba59dbd0d28bf68e64a42d7f2083c836f60f8e98fa"
+        }
+      ]
+    },
+    {
+      "name": "libgnomecanvas",
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://download.gnome.org/sources/libgnomecanvas/2.30/libgnomecanvas-2.30.3.tar.gz",
+          "sha256": "a8ca85e734ab03ecf1fba7d99e01ae2541d0df539c69a7da9414cde928c116da"
+        }
+      ]
+    },
+    {
+      "name": "poppler",
+      "buildsystem": "cmake-ninja",
+      "config-opts": [
+        "-DCMAKE_INSTALL_LIBDIR=/app/lib",
+        "-DCMAKE_INSTALL_INCLUDEDIR=/app/include",
+        "-DENABLE_LIBOPENJPEG=none"
+      ],
+      "cleanup": [
+        "/bin"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://poppler.freedesktop.org/poppler-0.64.0.tar.xz",
+          "sha256": "b21df92ca99f78067785cf2dc8e06deb04726b62389c0ee1f5d8b103c77f64b1"
+        }
+      ]
+    },
+    {
+      "name": "xournal",
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://downloads.sourceforge.net/xournal/xournal-0.4.8.2016.tar.gz",
+          "sha256": "b25898dbd7a149507f37a16769202d69fbebd4a000d766923bbd32c5c7462826"
+        }
+      ]
+    },
+    {
+      "name": "desktop-file",
+      "buildsystem": "simple",
+      "build-commands": [
+        "install -Dm644 xournal.desktop /app/share/applications/net.sourceforge.xournal.desktop"
+      ],
+      "sources": [
+        {
+          "type": "archive",
+          "url": "https://downloads.sourceforge.net/xournal/xournal-0.4.8.2016.tar.gz",
```

Comment:
You're downloading the source twice just to get the desktop file? You can add that somewhere else instead :)

Reply:
Sorry for being a noob, but how can I do this? 🙈

Reply:
By taking a look at the Makefile it seems that the desktop file is correctly installed 
`	$(INSTALL_DATA) $(srcdir)/xournal.desktop $(DESTDIR)$$desktopdir/applications; ` so you won't need that module anymore. You just have to patch the Makefile.am file to fix the prefix. As they're forcing either `/usr/`, `/usr/local` or `$HOME`. Flatpak uses `/app` instead

Reply:
Probably a stupid question but how do I create a .patch file?

Reply:
I tried adding a Makefile.patch file, which I generated with git diff before, but that somehow isn't working. Anything I missed in the process?